### DISCUSSION TITLE
为你的 Astro 博客添加评论功能

### DISCUSSION THREAD

**Discussion by giscus** [liruifengv/liruifengv.com#21] [Announcements]
# 为你的 Astro 博客添加评论功能

使用 giscus 为你的 Astro 博客添加评论功能。

http://localhost:4321/posts/add-comments-to-astro

**Comment by liruifengv** [comment]
测试评论

## 支持markdown

## 支持代码片段
```rs
use std::collections::HashMap;

fn two_sum(nums: &[i32], target: i32) -> Vec<i32> {
    let mut num_map: HashMap<i32, i32> = HashMap::new();

    for (i, &num) in nums.iter().enumerate() {
        if let Some(&j) = num_map.get(&(target - num)) {
            return vec![j, i as i32];
        }
        num_map.insert(num, i as i32);
    }

    vec![]
}
```

**Comment by BarrySong97** [comment]
这个组件很好用，但是必须要用户必须要科学上网才行

**Reply by BarrySong97** [reply]
不过国内的就在国内平台访问内容就好，个人网站还是针对开发者

**Comment by Liumingxun** [comment]
问一下，具体来说什么时候适合用`client:only`，什么时候更适合用`client:load`呢？ :thinking:

**Reply by liruifengv** [reply]
`client:load` 会在服务器端预渲染 dom 结构。
在某些情况下，预渲染会有问题，比如客户端组件使用了 esm 的包，预渲染中是cjs 就会报错。这时候就用 `client:only`。

如果不想考虑太多，可以无脑 `client:only`
